### PR TITLE
skip relative paths in forceInclude

### LINKED ISSUE #7822: "Force Include" config UI should not report validation errors.  Whether a forced include can be resolved is unknown until compiler is queried.
Bug type: Language Service

**Describe the bug**
- OS and Version: Windows
- C/C++ Extension Version: 1.5.1

**Steps to reproduce**

1. Enter an invalid path in "Force include" property in IntelliSense config UI. There should be error saying cannot find path.
2. Open the c_cpp_properties.json file
3. Observe there is not error in json file for invalid path
image

**Expected behavior**
UI and json file should show the same errors if the current selected active configuration is the one also show on config UI

### PR DESCRIPTION
**Pull request by elahehrashedi**
bug fix: https://github.com/microsoft/vscode-cpptools/issues/7822

### CODE REVIEW

**Review by sean-mcmanus — APPROVED** [review]

File: Extension/src/LanguageServer/configurations.ts [hunk 1]
```diff
@@ -1440,7 +1440,9 @@ export class CppProperties {
 
             // Check if resolved path exists
             if (!fs.existsSync(resolvedPath)) {
-                if (!this.rootUri) {
+                if (skipRelativePaths) {
```
Comment (bobbrow) [inline Extension/src/LanguageServer/configurations.ts]: This doesn't check that the resolved path is relative though.

Comment (bobbrow) [reply Extension/src/LanguageServer/configurations.ts]: It could be an absolute path that failed the 'exists' test

Comment (bobbrow) [reply Extension/src/LanguageServer/configurations.ts]: Try this: https://nodejs.org/api/path.html#path_path_isabsolute_path

File: Extension/src/LanguageServer/configurations.ts [hunk 2]
```diff
@@ -1622,13 +1624,22 @@ export class CppProperties {
...
                     for (const curPath of pathArray) {
                         paths.push(`${curPath}`);
                     }
                 }
             }
+            // Skip the relative forcedInclude files.
+            if (currentConfiguration.forcedInclude) {
+                for (const file of currentConfiguration.forcedInclude) {
+                    const resolvedFilePath: string = this.resolvePath(file, isWindows);
+                    if (util.checkFileExistsSync(resolvedFilePath) || util.checkDirectoryExistsSync(resolvedFilePath)) {
```
Comment (bobbrow) [inline Extension/src/LanguageServer/configurations.ts]: You don't need to check it exists here.  Use this API instead to decide: https://nodejs.org/api/path.html#path_path_isabsolute_path

**Review by bobbrow — CHANGES_REQUESTED** [review]

**Review by bobbrow — APPROVED** [review]

**Review by michelleangela — APPROVED** [review]

**Pull request merged**